x/auth/ante/sigverify.go

```diff
@@ -316,6 +319,16 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 		if !simulate && !ctx.IsReCheckTx() {
 			anyPk, _ := codectypes.NewAnyWithValue(pubKey)
 
+			// TODO HV2: do we need to add here (and modify accordingly) the following code from heimdall?
+			// in that case, we need to implement the RecoverPubkey method
+			/*
+				p, err := authTypes.RecoverPubkey(signBytes, sig.Bytes())
+				if err != nil {
+					return nil, sdk.ErrUnauthorized("signature verification failed; verify correct account sequence and chain-id").Result()
+				}
+				copy(pk[:], p[:])
+			*/
+
```

Comment:
I've been looking at the changes in [heimdall v1](https://github.com/Raneet10/cosmos-sdk/pull/2). Here are my observations:
* I have the impression that the code will somehow be needed. I understand `RecoverPubKey` is obtaining the public key according to the eth-go library, so if it was needed back then, I don't see why such "translation" wouldn't be needed now
* However, there's one fundamental difference in the base (vanilla) SDK code between `v0.37.4` and `v0.50.x`: the former is setting the public key if `nil`, whereas the latter is _expecting_ it to be already set (check lines 297-300 above). So if we need to uncomment the commented code in Heimdall v2, we will need to locate where the public key is first set on an account (and it's not here, unlike in `v0.37.4`).
* The good news is that this code is hit every time TX signatures are verified, so it is über frequent: if the commented code is indeed needed, something will break very early on in the game (IOW, it won't be a corner case we may miss)
* So I suggest to leave it as you have added it for now: as a comment, and wait for some test to show that it is needed (or not!)

Reply:
Resolved